File: app/src/main/java/closer/vlllage/com/closer/handler/data/PhoneContactsHandler.java
package closer.vlllage.com.closer.handler.data;

import java.util.List;

import closer.vlllage.com.closer.handler.group.PhoneContact;
import closer.vlllage.com.closer.handler.group.PhoneContacts;
import closer.vlllage.com.closer.handler.helpers.ApplicationHandler;
import closer.vlllage.com.closer.pool.PoolMember;
import io.reactivex.Observable;

public class PhoneContactsHandler extends PoolMember {

    private PhoneContacts phoneContacts;

    @Override
    protected void onPoolInit() {
        phoneContacts = $(ApplicationHandler.class).getApp().$(PhoneContacts.class);
    }

    public Observable<List<PhoneContact>> getAllContacts() {
        return phoneContacts.getAllContacts();
    }

    public void forceReload() {
        phoneContacts.forceReload();
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/data/PhoneContactsHandler.kt
package closer.vlllage.com.closer.handler.data

import closer.vlllage.com.closer.handler.group.PhoneContact
import closer.vlllage.com.closer.handler.group.PhoneContacts
import closer.vlllage.com.closer.handler.helpers.ApplicationHandler
import com.queatz.on.On
import com.queatz.on.OnLifecycle
import io.reactivex.Observable

class PhoneContactsHandler constructor(private val on: On) : OnLifecycle {

    private lateinit var phoneContacts: PhoneContacts

    val allContacts: Observable<List<PhoneContact>>
        get() = phoneContacts.allContacts

    override fun on() {
        phoneContacts = on<ApplicationHandler>().app.on<PhoneContacts>()
    }

    fun forceReload() {
        phoneContacts.forceReload()
    }
}
